### PR TITLE
[model] chore: add Qwen3.8-27B Megatron GRPO script

### PR DESCRIPTION
**Pull request by Feng0w0**
### What does this PR do?

> add Qwen3.8-27B Megatron GRPO script

### Checklist Before Starting

- [ ] Search for similar PRs. Paste at least one query link here: ...
- [ ] Format the PR title as `[{modules}] {type}: {description}` (This will be checked by the CI)
  - `{modules}` include `fsdp`, `megatron`, `veomni`, `sglang`, `vllm`, `rollout`, `trainer`, `ci`, `training_utils`, `recipe`, `hardware`, `deployment`, `ray`, `worker`, `single_controller`, `misc`, `perf`, `model`, `algo`, `env`, `tool`, `ckpt`, `doc`, `data`, `cfg`, `reward`, `fully_async`, `one_step_off`
  - If this PR involves multiple modules, separate them with `,` like `[megatron, fsdp, doc]`
  - `{type}` is in `feat`, `fix`, `refactor`, `chore`, `test`
  - If this PR breaks any API (CLI arguments, config, function signature, etc.), add `[BREAKING]` to the beginning of the title.
  - Example: `[BREAKING][fsdp, megatron] feat: dynamic batching`

### Test

We ran for 24 hours using the geo3k dataset, and the reward curves on both NPU and GPU showed an increase, with an error of less than 0.02 between the two devices

image
image

performance：
image

### API and Usage Example

> Demonstrate how the API changes if any, and provide usage example(s) if possible.

```python
# Add code snippet or script demonstrating how to use this
```

### Design & Code Changes

> Demonstrate the high-level design if this PR is complex, and list the specific changes.

### Checklist Before Submitting

> [!IMPORTANT]
> Please check all the following items before requesting a review, otherwise the reviewer might deprioritize this PR for review.

- [ ] Read the [Contribute Guide](https://github.com/verl-project/verl/blob/main/CONTRIBUTING.md).
- [ ] Apply [pre-commit checks](https://github.com/verl-project/verl/blob/main/CONTRIBUTING.md#code-linting-and-formatting): `pre-commit install && pre-commit run --all-files --show-diff-on-failure --color=always`
- [ ] Add / Update [the documentation](https://github.com/verl-project/verl/tree/main/docs).
- [ ] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/verl-project/verl/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: ...
- [ ] Once your PR is ready for CI, send a message in [the `ci-request` channel](https://verl-project.slack.com/archives/C091TCESWB1) in [the `verl` Slack workspace](https://join.slack.com/t/verl-project/shared_invite/zt-3855yhg8g-CTkqXu~hKojPCmo7k_yXTQ). (If not accessible, please try [the Feishu group (飞书群)](https://applink.larkoffice.com/client/chat/chatter/add_by_link?link_token=772jd4f1-cd91-441e-a820-498c6614126a).)
- [ ] If your PR is related to the `recipe` submodule, please also update the reference to the submodule commit via `git submodule update --remote` or `cd recipe && git pull origin main`.

### CODE REVIEW

File: examples/grpo_trainer/run_qwen3_8_27b_megatron.sh [hunk 1]
```diff
@@ -0,0 +1,247 @@
+#!/usr/bin/env bash
+# Qwen3.8-27B GRPO with Megatron/MindSpeed-Bridge and vLLM rollout.
+#
+# The default Ascend setup uses one 16-NPU node and the Geo3K dataset:
+#   TP=2 PP=4 CP=1 EP=1 ETP=1, rollout TP=8.
+#
```
Comment (beirong8kmiles) [inline examples/grpo_trainer/run_qwen3_8_27b_megatron.sh:6]: 是基于哪个镜像做的修改？是否可以和主线的镜像合并？

File: examples/grpo_trainer/run_qwen3_8_27b_megatron.sh [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+# Requirements on Ascend:
+#   - 8 A3 cards (2*64GB each, exposed as 16 NPU devices)
+#   - Base environment: Python 3.12.13, CANN 9.0.0, ATB 9.0.0.B160,
+#     PyTorch 2.10.0 and torch-npu 2.10.0.post4
+#   - Main Python packages:
+#       transformers==5.5.4 vllm==0.23.0 vllm-ascend==0.23.0 ray==2.56.1
+#       flash-linear-attention==0.5.2 fla-npu==1.0.0 nvidia-modelopt==0.45.0
+#   - Megatron-LM / Megatron-Core==0.18.0 (ba7b5ebce12a)
+#   - MindSpeed-Bridge==0.3.1 with Qwen3.8 provider registration
+#   - Megatron-Bridge==0.5.0 (e1ef727af058)
```
Comment (beirong8kmiles) [inline examples/grpo_trainer/run_qwen3_8_27b_megatron.sh:16]: 更新到支持模型列表中：https://github.com/verl-project/verl/blob/main/docs/ascend_tutorial/zh/model_support/model_and_algorithm_support.md

File: examples/grpo_trainer/run_qwen3_8_27b_megatron.sh [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+
+########################### Quick Config ###########################
+
+# DEVICE is auto-detected by probing torch_npu. Override it only when needed.
+DEVICE=${DEVICE:-$(python3 -c 'import torch_npu' 2>/dev/null && echo npu || echo gpu)}
+case "${DEVICE}" in
+    gpu)
+        TP=${TP:-2}
+        PP=${PP:-1}
+        CP=${CP:-1}
```
Comment (wucong25) [inline examples/grpo_trainer/run_qwen3_8_27b_megatron.sh:37]: 这里gpu也跑过么

File: examples/grpo_trainer/run_qwen3_8_27b_megatron.sh (hunk 1 above)
Comment (Feng0w0) [reply examples/grpo_trainer/run_qwen3_8_27b_megatron.sh:6]: 基于verl主线代码安装环境验证的，应该可以合并

File: examples/grpo_trainer/run_qwen3_8_27b_megatron.sh (hunk 2 above)
Comment (Feng0w0) [reply examples/grpo_trainer/run_qwen3_8_27b_megatron.sh:16]: 已添加

**Review by wucong25 — APPROVED** [review]

File: examples/grpo_trainer/run_qwen3_8_27b_megatron.sh (hunk 3 above)
Comment (Feng0w0) [reply examples/grpo_trainer/run_qwen3_8_27b_megatron.sh:37]: gpu没有跑过，但是该模型和qwen3.5结构一模一样，qwen3.5在GPU上跑过

**Review by wucong25 — APPROVED** [review]
